scripts/abi_artifact_generator/templates/printers.inc.jinja_template

```diff
@@ -0,0 +1,166 @@
+           register_list[index],
+           word_printing_helper->v);
+    printf("        Bytes: ");
+    PRINT_BYTES(/*( register_type )*/, word_printing_helper);
+  }
+
+  /* Dump Stack */
+  printf("    Stack: ");
+  typedef uint8_t stack_state[/*( stack_byte_count )*/];
+  stack_state *state_ptr = (stack_state *) (randomized_state + /*( register_count * register_size )*/);
+  MAKE_PRINT_HELPER(stack_state, state_ptr, stack_printing_helper);
+  PRINT_BYTES(stack_state, stack_printing_helper);
+}
+
+#define PRINT(TYPE, POINTER)                        \
+  do {                                              \
+    printf("      - Type: " #TYPE " # size = %zu\n" \
+           "        Bytes: ",                       \
+           sizeof(TYPE));                           \
+    MAKE_PRINT_HELPER(TYPE, POINTER, local_helper); \
+    PRINT_BYTES(TYPE, local_helper);                \
+  } while (0)
+
+#define POINTER(TYPE, POINTER)                        \
+  do {                                                \
+    printf("      - Type: " #TYPE " # size = %zu\n"   \
+           "        Pointer: 0x%zx\n        Bytes: ", \
+           sizeof(TYPE),                              \
+           (/*( register_type )*/) POINTER);          \
+    MAKE_PRINT_HELPER(TYPE *, POINTER, local_helper); \
+    PRINT_BYTES(TYPE *, local_helper);                \
+  } while (0)
+
+/*{ for name, arguments in argument_functions.items() -}*/
+NOINLINE void ABIDEF print_/*( name )*/(void) {
+  puts("    Function: \"/*( "test_" + name )*/\"");
+  puts("    Arguments:");
+
+  CurrentOffset = 0;
+
+  /*{- for type in arguments }*/
+  /*( type )*/ argument_/*( loop.index - 1 )*/; (void) argument_/*( loop.index - 1 )*/;
+  if (sizeof(/*( type )*/) <= ARGUMENT_SIZE_LIMIT(argument_/*( loop.index - 1 )*/) && !IS_PACKED(/*( type )*/)) {
+    PRINT(/*( type )*/, (/*( type )*/ *) ((uint8_t *) TemporaryLocation + CurrentOffset));
+    CurrentOffset += sizeof(/*( type )*/);
+  } else {
+    POINTER(/*( type )*/, (/*( type )*/ **) ((uint8_t *) TemporaryLocation + CurrentOffset));
+    CurrentOffset += sizeof(/*( type )*/ *);
+  }
+/*{ endfor }*/
+  puts("    ReturnValue:");
+  puts("");
+}
+
+/*{ endfor -}*/
+
+static jmp_buf jump_buffer;
+/*{ for name, return_value_type in return_value_functions.items() -}*/
+NOINLINE void print_/*( name )*/(void) {
+  puts("    Function: \"/*( "test_" + name )*/\"");
+  puts("    Arguments:");
+  puts("    ReturnValue:");
+
+  /*( return_value_type )*/ return_value;
+  if (sizeof(return_value) <= RETURN_VALUE_SIZE_LIMIT(return_value) && !IS_PACKED(/*( return_value_type )*/)) {
+    PRINT(/*(- return_value_type )*/, (/*(- return_value_type )*/ *) TemporaryLocation);
+  } else {
+    POINTER(/*(- return_value_type )*/, (/*(- return_value_type )*/ **) TemporaryLocation);
+  }
+
+  puts("");
+}
+
+/*{ endfor -}*/
+
+uint8_t TemporaryData[/*( generated_byte_count )*/];
+void *TemporaryLocation = &TemporaryData;
+ptrdiff_t CurrentOffset = 0;
```

Comment:
Docs

Reply:
Closing all the doc-related threads for now.